src/typed_array.rs

```diff
@@ -0,0 +1,360 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::{sys, JSException, JSObject, JSTypedArray, JSTypedArrayType, JSValue};
+use std::{ptr, slice};
+
+impl JSTypedArray {
+    /// Create a new [`Self`] from its raw pointer directly.
+    ///
+    /// # Safety
+    ///
+    /// Ensure `raw` is valid.
+    pub(crate) unsafe fn from_raw(ctx: sys::JSContextRef, raw: sys::JSObjectRef) -> Self {
+        Self { ctx, raw }
+    }
+
+    /// Returns a value of type [`JSTypedArrayType`] that identifies value's
+    /// Typed Array type, or `JSTypedArrayType::None` if the value is not a Typed Array
+    /// object.
+    ///
+    /// ```rust
+    /// # use javascriptcore::*;
+    /// let ctx = JSContext::default();
+    /// let array = evaluate_script(
+    ///     &ctx,
+    ///     "new Uint8Array([1, 2, 3, 4, 5])",
+    ///     None,
+    ///     "foo.js",
+    ///     1,
+    /// )
+    ///     .unwrap()
+    ///     .as_typed_array()
+    ///     .unwrap();
+    /// assert_eq!(array.ty().unwrap(), JSTypedArrayType::Uint8Array);
+    /// ```
+    pub fn ty(&self) -> Result<JSTypedArrayType, JSException> {
+        let mut exception: sys::JSValueRef = ptr::null_mut();
+        let value = unsafe { sys::JSValueGetTypedArrayType(self.ctx, self.raw, &mut exception) };
+
+        if !exception.is_null() {
+            Err(unsafe { JSValue::from_raw(self.ctx, exception) }.into())
+        } else {
+            Ok(value)
+        }
+    }
+
+    /// Returns the length of the Typed Array.
+    ///
+    /// ```rust
+    /// # use javascriptcore::*;
+    /// let ctx = JSContext::default();
+    /// let array = evaluate_script(
+    ///     &ctx,
+    ///     "new Uint8Array([1, 2, 3, 4, 5])",
+    ///     None,
+    ///     "foo.js",
+    ///     1,
+    /// )
+    ///     .unwrap()
+    ///     .as_typed_array()
+    ///     .unwrap();
+    /// assert_eq!(array.len().unwrap(), 5);
+    /// ```
+    #[allow(clippy::len_without_is_empty)]
```

Comment:
Just add an is_empty?

Reply:
I wasn't sure it was making sense to have `TypedArray::is_empty`, because all items exists by definition, just with a default value, so a typed array is _never_ empty.